libs/tunables/c_generator.py

```diff
@@ -0,0 +1,123 @@
+from jinja2 import Environment, FileSystemLoader
+import sys
+import yaml
+import os
+
+# Currently running this in oem directory
+template_path = "libs/tunables/templates"
+
+hfile = "h_file.j2"
+cfile = "c_file.j2"
+
+
+def main():
+    """CLI that generates C files from a Yaml file."""
+
+    ecu_list = sys.argv
+
+    # Currently this can generate the c structs for multiple boards at once.
+    # The format would be: python c_generator.py ecu1 ecu2 ecu3
+    for ecu in ecu_list[1:]:
+        ecu_path = "vehicle/mkv/software/{}".format(ecu)
+        ymlpath = ecu_path + "/tunables.yml"
+
+        # Check that the path to the yml file is correct
+        if os.path.exists(ymlpath):
+
+            # Reads the tunables.yml file & stores all the struct variables
+            paramslist = get_params(ymlpath)
+            # Generates a C struct with all the parameters in the bms board
+            c_content = c_maker(ecu, paramslist)
+            h_content = h_maker(ecu)
+
+            # What we made via the jinja template.
+            c_out = "{}/tunables.c".format(ecu_path)
+            h_out = "{}/tunables.h".format(ecu_path)
+
+            # Write the .c and .h files
+            with open(c_out, "w+") as file:
+                file.write(c_content)
+                file.close()
+
+            with open(h_out, "w+") as file:
+                file.write(h_content)
+                file.close()
+
+            # Check that we wrote everything correctly
+            print(f"\n ########## {ecu} tunables.c file ########## \n")
+            with open(c_out, "r") as file:
+                print(file.read())
+                file.close()
+
+            print(f"\n ########## {ecu} tunables.h file ########## \n")
+            with open(h_out, "r") as file:
+                print(file.read())
+                file.close()
+
+        else:
+            # Error msg
+            print(f"ERROR: tunables.yml at {ymlpath} does not exist")
```

Comment:
Didn't raise an exception because I don't want to break the for loop. 

For example, the line:

python c_generator.py throttle fakeECU bms

This should generate the .c and .h files for throttle and bms but since fakeECU isn't a real ECU, it'll just notify the user and then move on.

Reply:
I would argue that this is not good behavior for a program. In my strong opinion, programs that run automatically (i.e. not directly by the user, as is the case with your code) should not be fault tolerant, and instead should fail loudly and explicitly when there is an error. This is so that your code can point to issues elsewhere. For example, if your code silently fails and moves on, we lose the ability to detect if the thing that called your code (i.e. the Bazel build system) has some error or misconfiguration.